javalib/src/main/scala/java/io/OutputStreamWriter.scala

```diff
@@ -45,7 +45,11 @@ class OutputStreamWriter(private[this] var out: OutputStream,
     this(out, Charset.defaultCharset)
 
   def this(out: OutputStream, charsetName: String) =
-    this(out, Charset.forName(charsetName))
+    this(out, try {
+      Charset.forName(charsetName)
+    } catch {
+      case _: UnsupportedCharsetException => throw new UnsupportedEncodingException(charsetName)
```

Comment:
Should we link the original exception as cause? (please reproduce JVM behavior if not specified, maybe even test it?)

Reply:
We cannot (therefore should not) link the original exception as cause, since [`UnsupportedEncodingException` does not have constructors to accept cause](https://docs.oracle.com/javase/8/docs/api/java/io/UnsupportedEncodingException.html).

Added tests to verify cause is `null` in https://github.com/scala-js/scala-js/pull/3687/commits/a1d0dce3e24e6580e8dfdd4b6854d446d47ed736